pkg/virt-handler/container-disk/mount.go

```diff
@@ -41,7 +41,7 @@ func (m *Mounter) Mount(vmi *v1.VirtualMachineInstance, verify bool) error {
 				if err != nil {
 					return fmt.Errorf("failed to detect root mount point of containerDisk %v on the node: %v", volume.Name, err)
 				}
-				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
+				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint, nodeMountInfo.Root), volume.ContainerDisk.Path)
 				if err != nil {
```

Comment:
The fix works, however we probably need switch `nodeMountInfo.MountPoint` and `nodeMountInfo.Root` to be 

```
sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.Root, nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
```

Reply:
I am not sure. You gave me that example: `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/rootfs/disk` This indicates that we first need to add the `MountPoint which is `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/`, then we need to add the Root which is `/rootfs` and then finally the `Path` from the containerdisk: `/disk`. 

If it works for you already, I would suspect that I would break it when I switch it. Can that be?

Reply:
@rmohr It won't break. the `nodeMountInfo.Root` is `record[3]`, while `nodeMountInfo.MountPoint` is `record[4]`  in `/proc/<pid>/mountinfo`. Therefore, the full path should be `nodeRes.MountRoot()`/`nodeMountInfo.Root`/`nodeMountInfo.MountPoint`/`volume.ContainerDisk.Path`.  In the case, as the `nodeMountInfo.MountPoint` is `/`, both actualy work if we switch `nodeMountInfo.Root` and `nodeMountInfo.MountPoint`. However logically the `nodeMountInfo.MountPoint` should be after `nodeMountInfo.Root`.

Reply:
I still don't get it, since the path says that `record[3]`, which is `/rootfs` is after `/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/` which is `record[4].

Maybe you could post again a mount table from the host perspective, so that we can verify thinks on a real mountinfo?

Reply:
This should be `filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint, mountInfo.Root)`. Pls see https://github.com/kubevirt/kubevirt/pull/3182

Reply:
@xuxinkun that can very well be correct, I would just prefer to see once a real mountinfo from the host perspective to verify this.

Reply:
@rmohr @xuxinkun In the logged case, `nodeRes.MountRoot()` is `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171`, `nodeMountInfo.Root` is `/rootfs`, `nodeMountInfo.MountPoint` is `/` and `volume.ContainerDisk.Path` is `disk`. 

We all know the right path should be `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/rootfs/disk`. Since `nodeMountInfo.MountPoint` is `/`, no matter `filepath.Join(nodeRes.MountRoot(), nodeMountInfo.Root, nodeMountInfo.MountPoint)` or `filepath.Join(nodeRes.MountRoot(), nodeMountInfo.MountPoint, mountInfo.Root)`, the result is always `/proc/1/root/opt/docker/devicemapper/mnt/83f8464c5ce21fa7334ed8d181d235c935c41421f347c2d03d3347abe2488171/rootfs/`. However, logically the `nodeMountInfo.MountPoint` should be after `nodeMountInfo.Root`. 
The problem of the original codes is missing the `mountInfo.Root` in the MIDDLE of the path.

Reply:
Back from PTO and updated finally. Can you review it again?